packages/client/utils/SlackManager.ts

```diff
@@ -52,14 +69,15 @@ export interface SlackPublicConversation {
 
 type SlackConversation = SlackPublicConversation | SlackIM
 
-interface SlackChannelInfo {
+export interface SlackChannelInfo {
   id: string
   name: string
   is_channel: boolean
   created: number
   is_archived: boolean
   is_general: boolean
   name_normalized: string
+  is_im: boolean
```

Comment:
this is a perfect place to test out writing your first discriminated union in typescript.
do a little digging on the subject & try it out.

Reply:
I realised that this interface was no longer being used so I've removed it but I got the chance to play with the discriminated unions with `SlackCoversation` and `ConversationListResponse.channels`! I've added type guard checks [here](https://github.com/ParabolInc/parabol/blob/slack-migration/packages/client/hooks/useSlackChannels.ts#L30) and [here](https://github.com/ParabolInc/parabol/blob/slack-migration/packages/server/graphql/mutations/helpers/notifySlack.ts#L129) to prevent recasting